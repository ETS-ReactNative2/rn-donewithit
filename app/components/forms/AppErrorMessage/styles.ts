import CONSTS from './consts'
import COLORS from '../../../config/colors'
import { StyleSheet } from 'react-native'
const styles = StyleSheet.create({
  text: {
    color: COLORS.PRIMARY,
    fontSize: CONSTS.FONT_SIZE,
    height: CONSTS.ERROR_HEIGHT,
    lineHeight: CONSTS.ERROR_HEIGHT,
    width: '100%'
  }
})

export default styles
