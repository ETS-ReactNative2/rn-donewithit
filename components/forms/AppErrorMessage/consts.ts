const CONSTS = {
  ERROR_HEIGHT: 30,
  FONT_SIZE: 18
}

export default CONSTS
