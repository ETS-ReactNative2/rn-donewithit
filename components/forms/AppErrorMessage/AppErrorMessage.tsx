import React from 'react'
import AppText from '../../AppText'
import styles from './styles'

function AppErrorMessage({ style, errorMessage, ...otherProps }: any) {
  return (
    <AppText style={[styles.text, style]} {...otherProps}>
      {errorMessage}
    </AppText>
  )
}

export default AppErrorMessage
