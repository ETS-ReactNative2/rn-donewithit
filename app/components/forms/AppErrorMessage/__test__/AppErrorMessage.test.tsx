import React from 'react'
import { StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import AppErrorMessage from '../index'

const customStyles = StyleSheet.create({
  errorMessage: {
    color: 'red',
    fontSize: 16
  }
})

const props = {
  style: customStyles.errorMessage,
  errorMessage: 'Error: test error message'
}

describe('AppErrorMessage', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppErrorMessage {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
