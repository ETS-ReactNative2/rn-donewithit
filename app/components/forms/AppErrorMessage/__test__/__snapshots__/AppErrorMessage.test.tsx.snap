// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppErrorMessage renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Courier",
        "fontSize": 13,
      },
      Array [
        Object {
          "color": "tomato",
          "fontSize": 18,
          "height": 30,
          "lineHeight": 30,
          "width": "100%",
        },
        Object {
          "color": "red",
          "fontSize": 16,
        },
      ],
    ]
  }
>
  Error: test error message
</Text>
`;
